build.gradle

```diff
@@ -44,3 +46,50 @@ dependencyUpdates.resolutionStrategy {
 		}
 	}
 }
+
+task buildMavenPlugin(type: Exec, dependsOn: ':agent:publishToMavenLocal') {
```

Comment:
@DreierF awesome! I was searching for something like this but failed to find anything. Thanks for digging this up!
Since this would mean removing the `pom.xml` but @sewe had some specific review comments for that, could you please clarify with @sewe that building with Gradle is our preferred solution? If you both agree, then I'll try to get one of the two plugins to work :-) I'd just like to avoid that I change the build to Gradle and then @sewe has some objections that can't be realized with the Gradle plugins.

Reply:
Just to be clear: I'd personally prefer building the Maven plugin with Gradle as well, since it makes our build process much easier to reason about - especially when the build fails. The current "Gradle calls Maven via the command line" solution is really rather hacky and probably masks errors too easily and makes debugging broken builds harder than necessary.

Reply:
If the build via Gradle adds some unintended restrictions it's fine for me to not use them, but I would still prefer to use https://github.com/dkorotych/gradle-maven-exec-plugin -> `MavenExec` as a replacement for `Exec`.

Reply:
What exactly would we miss by moving to a Gradle based solution?

Reply:
Based on past experiences, I am strongly oppossed to not using the "native" build solution for _any_ kind of plug-in. That's why we build, for example, our IntelliJ plug-in with Gradle, our Eclipse plug-in with Maven+Tycho and our NetBeans plug-in with plain Maven; anything else is just more painful than it needs to be. (FWIW, we did have a homegrown Gradle build for the NetBeans plug-in at one point, but lost a lot of IDE support like debugging the plug-in due to this.)

> What exactly would we miss by moving to a Gradle based solution?

Well, in audit terms, you would IMHO rightfully earn a yellow assesment for criteria like **vendor support** (the "native" build solution by definition will be maintained as long as the plug-in's target will be maintained) and **tool support** (as typically 99.9% of plug-ins will be build using the native solution and you get better IDE support, e.g. using [Maven Development Tools](https://marketplace.eclipse.org/content/maven-development-tools-requires-m2e-16) which can interpret a `maven-plugin-plugin`-based build but not one based on some not very widespread Gradle plugin).

Also, any newcomer to the build will be able to figure it out with a bit of Googling `"maven-plugin-plugin examples"` if you use the default build solution, but for any custom solution you would need to understand (in this case) Gradle and a far less widely used Gradle plug-in.

But let me turn @DreierF's question on its head: "What would we gain by moving to a Gradle-based solution"?

AFAICT, all you would gain is that you have a single `gradlew` call rather than a `gradlew publishToMavenLocal` followed by `mvnw` invocation in your `actions.yml`. And that is not a very convincing reason to stray for the default path, IMHO. 

Moreover, workarounds like this would not be necessary: https://github.com/cqse/teamscale-jacoco-agent/blob/6a3ce80e71c663a3dbd30fadc342f19ddc9f05e7/teamscale-surefire-provider/build.gradle#L43

So, please build the [`teamscale-gradle-plugin`](https://github.com/cqse/teamscale-jacoco-agent/tree/6a3ce80e71c663a3dbd30fadc342f19ddc9f05e7/teamscale-gradle-plugin) with Gradle and the [`teamscale-maven-plugin`](https://github.com/cqse/teamscale-jacoco-agent/tree/6a3ce80e71c663a3dbd30fadc342f19ddc9f05e7/teamscale-maven-plugin) with Maven (and use `publishToMavenLocal` as the glue between the Gradle and Maven worlds). :pray:

Reply:
Thanks for clarifying between you two which way to go. Will keep it as is now and try to use the MavenExec thing. Just for future reference: the above comment has nothing to do with using Gradle to build a Maven plugin, this an artifact of shadowing the surefire provider (which is not a Maven plugin but simply a plain old Jar file containing a service provider file).

Reply:
@DreierF the MavenExec plugin seems to be dead (no commits in over a year) and doesn't work with Gradle 7. So I'll stick with simple Exec tasks then